postgresql/database.go

```diff
@@ -59,6 +59,10 @@ type cachedStatement struct {
 	query string
 }
 
+func (c *cachedStatement) OnPurge() {
+	c.Stmt.Close()
```

Comment:
You need to handle the error returned by close: https://github.com/golang/go/blob/0104a31b8fbcbe52728a08867b26415d282c35d2/src/database/sql/driver/driver.go#L118-L120